src/iohandler/process_io_handler.cc

```diff
@@ -139,17 +139,9 @@ void ProcessIOHandler::open(enum UpnpOpenFileMode mode)
     }
 
     if (mode == UPNP_READ)
-#ifdef __linux__
-        fd = ::open(filename.c_str(), O_RDONLY | O_NONBLOCK | O_CLOEXEC);
-#else
-        fd = ::open(filename.c_str(), O_RDONLY | O_NONBLOCK);
-#endif
+        fd = ::open(filename.c_str(), linux ? (O_RDONLY | O_NONBLOCK | O_CLOEXEC) : (O_RDONLY | O_NONBLOCK));
```

Comment:
Is `O_CLOEXEC` define on non-linux system?

Reply:
I don't think so.

Reply:
Yes, because it's only running on Linux-like systems. Some users still compile for Sun etc.

Reply:
shouldn't be an issue. all of this should be done at compile time.